### PR TITLE
Update Node.js

### PR DESCRIPTION
**Pull request by labriola**
Inverted hasChildNodes() logic to ensure that the existance of a firstChild means there are childNodes

### CODE REVIEW

**Comment by arv** [comment]
Thanks,

2 things.
1. Did you sign the CLA? https://github.com/Polymer/polymer/blob/master/CONTRIBUTING.md
2. Do you mind adding a test to make sure I don't break this again?

**Comment by labriola** [comment]
CLA is signed and filed.

I will add the test next and get this submitted too.

**Comment by labriola** [comment]
Test added to this branch. Want me to open a new pull request or sufficient?

File: test/js/Node.js [hunk 1]
```diff
@@ -186,4 +186,19 @@ suite('Node', function() {
 
     expectStructure(host2, {});
   });
+  
+  test('hasChildNodes without a shadow root', function() {
+    var div = document.createElement('div');
+    
+    //No children yet
+    assert( !div.hasChildNodes(), "should be false with no children" );
```
Comment (arv) [inline test/js/Node.js]: I'm going to be a bit nit picky here. We follow http://google-styleguide.googlecode.com/svn/trunk/javascriptguide.xml
- No space after (
- No space before )
- Use single quotes

File: test/js/Node.js [hunk 2]
```diff
@@ -186,4 +186,19 @@ suite('Node', function() {
...
+    
+    //Add a child
```
Comment (arv) [inline test/js/Node.js]: Space after //

File: test/js/Node.js (hunk 1 above)
Comment (arv) [inline test/js/Node.js]: Use assert.isTrue instead so that this fails for thruthy values that are not boolean

**Comment by labriola** [comment]
No problem at all. I will fix these and resubmit.

If there is anything else just let me know as I intend on submitting a bunch of small patches over the next few weeks so any time we spend now will likely reduce time later.

**Comment by labriola** [comment]
Okay, updates. I ended up using assert.isTrue() and assert.isFalse(). Wasn't sure from your comment if that is acceptable or if you only wanted me to you isTrue(). Let me know if you want this revised and I will take a read through your code guidelines. Thanks.

File: test/js/Node.js [hunk 3]
```diff
@@ -186,4 +186,19 @@ suite('Node', function() {
 
     expectStructure(host2, {});
   });
+  
+  test('hasChildNodes without a shadow root', function() {
+    var div = document.createElement('div');
+    
+    // No children yet
```
Comment (arv) [inline test/js/Node.js]: No need for these comments. Comments should explain why and not what.

**Comment by arv** [comment]
LGTM

Can you squash these into a single commit?

(there might be one more churn tomorrow, I have to look into our policy related to the AUTHORS file)

**Comment by labriola** [comment]
Sure, just let me know tomorrow after you check on the final details and I will wrap it up into a single commit and get it through.

**Comment by labriola** [comment]
Now a single commit

**Comment by arv** [comment]
Thanks a lot

**Pull request merged**